File: new Qs in 30 days challenge/Counting Elements.py
from collections import defaultdict, Counter
class Solution:
    def countElements(self, arr: List[int]) -> int:
        # s = set()
        if not arr:
            return 0
        
        c_map = Counter(arr)
        keys, counts = list(c_map.keys()), list(c_map.values())
        res = 0
        for i, k in enumerate(keys):
            if k+1 in keys:
                res += counts[i]

        return res

File: new Qs in 30 days challenge/Counting Elements.js
var countElements = function(arr) {    
    let d = {}
    for (let i=0; i < arr.length; i++){
        if (arr[i] in d){
            d[arr[i]] += 1;
        }
        else {
            d[arr[i]] = 1;
        }
    }
    
    let res = 0
    for (let key in d){
        // if ((key+1) in d){
        if (parseFloat(key)+1 in d){
            res += d[key];
        }
    }
    return res;
};

console.log(countElements([1,2,3]));